### PR TITLE
Fixing Action imports and defaults AnomalyResult grade to 0 if negative

### PR DESCRIPTION
**Pull request by joshpalis**
### Description
Resolves compilation errors coming from upstream refactorization and resolves negative Anomaly grade issue by defaulting grade to 0 if negative

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by joshpalis** [comment]
BWC Tests failing due to : 
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':adBwcCluster#fullRestartClusterTask'.
> Can't get https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.10.0/latest/linux/x64/tar/builds/opensearch/plugins/opensearch-anomaly-detection-2.10.0.0.zip to /home/runner/work/anomaly-detection/anomaly-detection/src/test/resources/org/opensearch/ad/bwc/anomaly-detection/2.10.0.0/opensearch-anomaly-detection-2.10.0.0.zip
```

Looking at the distribution builds, there hasn't been a successful 2.10.0 build so this error is due to stale snapshots. @kaituo do we want to block this PR until the snapshots are updated?

**Review by kaituo — APPROVED** [review]

**Comment by kaituo** [comment]
> BWC Tests failing due to :
> 
> ```
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Could not determine the dependencies of task ':adBwcCluster#fullRestartClusterTask'.
> > Can't get https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.10.0/latest/linux/x64/tar/builds/opensearch/plugins/opensearch-anomaly-detection-2.10.0.0.zip to /home/runner/work/anomaly-detection/anomaly-detection/src/test/resources/org/opensearch/ad/bwc/anomaly-detection/2.10.0.0/opensearch-anomaly-detection-2.10.0.0.zip
> ```
> 
> Looking at the distribution builds, there hasn't been a successful 2.10.0 build so this error is due to stale snapshots. @kaituo do we want to block this PR until the snapshots are updated?

No, we don't want to.

File: src/main/java/org/opensearch/ad/model/AnomalyResult.java [hunk 1]
```diff
@@ -398,7 +398,7 @@ public static AnomalyResult fromRawTRCFResult(
             detectorId,
             taskId,
             rcfScore,
-            grade,
+            Math.max(0, grade),
```
Comment (sudiptoguha) [inline src/main/java/org/opensearch/ad/model/AnomalyResult.java:401]: :)

**Review by sudiptoguha — APPROVED** [review]

**Comment by joshpalis** [comment]
Will raise a manual backport, sans import changes

**Pull request merged**